ch_tools/chadmin/cli/table_metadata_parser.py

```diff
@@ -0,0 +1,96 @@
+import re
+import uuid
+from enum import Enum
+
+UUID_PATTERN = "UUID"
+ENGINE_PATTERN = "ENGINE"
+
+
+class MergeTreeFamilyEngines(Enum):
+    MERGE_TREE = "MergeTree"
+    REPLACING_MERGE_TREE = "ReplacingMergeTree"
+    SUMMING_MERGE_TREE = "SummingMergeTree"
+    AGGREGATING_MERGE_TREE = "AggregatingMergeTree"
+    COLLAPSING_MERGE_TREE = "CollapsingMergeTree"
+    VERSIONED_MERGE_TREE = "VersionedCollapsingMergeTree"
+    REPLICATED_MERGE_TREE = "ReplicatedMergeTree"
+
+    @staticmethod
+    def from_str(engine_str: str) -> "MergeTreeFamilyEngines":
+        for engine in MergeTreeFamilyEngines:
+            if engine.value == engine_str:
+                return engine
+        raise RuntimeError(
+            f"Doesn't support removing detached table with engine {engine_str}"
+        )
+
+
+class TableMetadataParser:
+    @property
+    def table_uuid(self):
+        return self._table_uuid
+
+    @property
+    def table_engine(self):
+        return self._table_engine
+
+    def __init__(self, table_metadata_path: str):
+        assert table_metadata_path.endswith(".sql")
+
+        self._table_uuid = None
+        self._table_engine = None
```

Comment:
still broken single resp
```suggestion
class TableMetadata:
       self.table_uuid = None
       self.table_engine = None
       

class TableMetadataParser:
    @property
    def get_table_metadata(self):
        return self._table_metadata

    def __init__(self, table_metadata_path: str):
        assert table_metadata_path.endswith(".sql")
        self._table_metadata
```

Reply:
If we are talking about the Single Responsibility Principle, the above option does not significantly differ from the original. At the very least, it still stores the same data, but with additional abstraction. In my opinion, a parser is not the type of task where this principle needs to be discussed. I know plenty of examples of parsers that either store or do not store data themselves. From the perspective of solving the current task, this is not important.

In any case, I have now implemented the classic approach.